src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -946,14 +946,48 @@ public BlockHeader[] FindHeaders(Keccak? blockHash, int numberOfBlocks, int skip
 
             if (skip == 0)
             {
+                static BlockHeader[] FindHeadersReversedFast(BlockTree tree, BlockHeader startHeader, int numberOfBlocks, bool reverse = false)
```

Comment:
Why move in?

Reply:
To be closer to only place where it is used.